main/adapters/http_server/http_server_test.go

```diff
@@ -2,11 +2,905 @@ package http_server
+			},
+		},
+		{
+			name: "chain",
+			request: func() *http.Request {
+				payload := []byte("{\"c\": \"d\", \"a\": \"b\", \"e\": \"f\"}")
+				request := httptest.NewRequest(http.MethodPost, "/5133fbdd-978d-4f95-9af9-41abdef2f2b4", bytes.NewReader(payload))
+				request.Header.Add(XAuthHeader, testAuth)
+				request.Header.Add("Content-Type", JSONType)
+				return request
+			}(),
+			setExpectations: func(m *mock.Mock) {
+				// checkAuth(ctx context.Context, uid uuid.UUID, auth string) (ok, found bool, err error)
+				m.On("checkAuth", mock.AnythingOfType("*context.timerCtx"), uuid.MustParse("5133fbdd-978d-4f95-9af9-41abdef2f2b4"), testAuth).Return(true, true, nil)
+				m.On("sign",
+					mock.AnythingOfType("*context.timerCtx"),
+					uuid.MustParse("5133fbdd-978d-4f95-9af9-41abdef2f2b4"),
+					testAuth,
+					Sha256Sum{0x80, 0xc9, 0x83, 0xc2, 0xfa, 0x61, 0x75, 0x1b, 0x2f, 0x78, 0x42, 0xa3, 0xa3, 0x39, 0x34, 0xfc, 0xbe, 0xd1, 0xc4, 0x3a, 0xa2, 0x5c, 0xa3, 0xb6, 0x39, 0x5c, 0x12, 0xf5, 0x53, 0xe2, 0xf0, 0x5e},
+					ChainHash,
+					false,
+				).Return(HTTPResponse{
+					StatusCode: http.StatusOK,
+					//Header:     http.Header{"test": []string{"1", "2", "3"}},
+					Header:  http.Header{"test": []string{"header"}},
+					Content: []byte("chained"),
+				})
+			},
+			tcChecks: func(t *testing.T, w *httptest.ResponseRecorder, m *mock.Mock) {
+				m.AssertExpectations(t)
+				assert.Equal(t, http.StatusOK, w.Code)
+				//assert.Equal(t, []string{"1", "2", "3"}, w.Header().Get("test"))
+				assert.Equal(t, "header", w.Header().Get("test"))
+				assert.Contains(t, w.Body.String(), "chained")
+			},
+		},
+		{
+			name: "chain hash",
+			request: func() *http.Request {
+				payload := []byte("gMmDwvphdRsveEKjozk0/L7RxDqiXKO2OVwS9VPi8F4=")
+				request := httptest.NewRequest(http.MethodPost, "/5133fbdd-978d-4f95-9af9-41abdef2f2b4/hash", bytes.NewReader(payload))
+				request.Header.Add(XAuthHeader, testAuth)
+				request.Header.Add("Content-Type", "text/plain")
+				return request
+			}(),
+			setExpectations: func(m *mock.Mock) {
+				m.On("checkAuth", mock.AnythingOfType("*context.timerCtx"), uuid.MustParse("5133fbdd-978d-4f95-9af9-41abdef2f2b4"), testAuth).Return(true, true, nil)
+				m.On("sign",
+					mock.AnythingOfType("*context.timerCtx"),
+					uuid.MustParse("5133fbdd-978d-4f95-9af9-41abdef2f2b4"),
+					testAuth,
+					Sha256Sum{0x80, 0xc9, 0x83, 0xc2, 0xfa, 0x61, 0x75, 0x1b, 0x2f, 0x78, 0x42, 0xa3, 0xa3, 0x39, 0x34, 0xfc, 0xbe, 0xd1, 0xc4, 0x3a, 0xa2, 0x5c, 0xa3, 0xb6, 0x39, 0x5c, 0x12, 0xf5, 0x53, 0xe2, 0xf0, 0x5e},
+					ChainHash,
+					false,
+				).Return(HTTPResponse{
+					StatusCode: http.StatusOK,
+					Header:     http.Header{"test": []string{"header"}},
+					Content:    []byte("chained hash"),
+				})
+			},
+			tcChecks: func(t *testing.T, w *httptest.ResponseRecorder, m *mock.Mock) {
+				m.AssertExpectations(t)
+				assert.Equal(t, http.StatusOK, w.Code)
+				assert.Equal(t, "header", w.Header().Get("test"))
```

Comment:
I don't get what this tests for. This happens a few more time, I will not make new comments for the other places.

Reply:
same as above